src/opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -342,73 +342,134 @@ Schedule::Schedule(const Deck& deck, const EclipseState& es, const std::optional
+                      });
     }
 
-    void info(const std::vector<std::string>& msg_list) {
-        for (const auto& msg : msg_list)
-            this->info(msg);
+    void info(const std::string& msg)
+    {
+        OpmLog::info(this->format_message(msg));
     }
 
-    void complete_step(const std::string& msg) {
-        this->step_count += 1;
-        if (this->step_count == this->max_print) {
-            this->log_function(this->prefix + msg);
-            this->info(std::vector<std::string>{"Report limit reached, see PRT-file for remaining Schedule initialization.", ""});
-            this->log_function = &OpmLog::note;
-        } else {
-            this->log_function( this->prefix + msg );
-            this->log_function( this->prefix );
+    void complete_step(const std::string& msg)
+    {
+        (*this)(msg);
+
+        this->step_count_ += 1;
+        if (this->step_count_ == this->max_print_) {
+            this->redirect(&OpmLog::note, {"Report limit reached, see PRT-file for remaining Schedule initialization.", ""});
+        }
+        else {
+            // Blank line
+            (*this)("");
         }
     };
 
-    void restart() {
-        this->step_count = 0;
-        this->log_function = &OpmLog::info;
+    void restart()
+    {
+        this->step_count_ = 0;
+        this->redirect(&OpmLog::info);
     }
 
-    void location(const KeywordLocation& location) {
-        if (this->current_file == location.filename)
+    void location(const KeywordLocation& location)
+    {
+        if (this->current_file_ == location.filename) {
             return;
+        }
 
-        this->operator()( fmt::format("Reading from: {} line {}", location.filename, location.lineno) );
-        this->current_file = location.filename;
+        (*this)(fmt::format("Reading from: {} line {}", location.filename, location.lineno));
+        this->current_file_ = location.filename;
     }
 
-
 private:
-    std::size_t step_count = 0;
-    std::size_t max_print  = 5;
-    std::string prefix;
-    std::string current_file;
-    void (*log_function)(const std::string&);
+    using LogFunction = void (*) (const std::string&);
+
+    Stream stream_{Stream::Info};
+    std::size_t step_count_ = 0;
+    std::size_t max_print_  = 5;
+    std::string prefix_{};
+    std::string current_file_{};
+
+    LogFunction log_function_{nullptr};
+
+    void redirect(const LogFunction               new_stream,
+                  const std::vector<std::string>& messages = {})
+    {
+        if (this->stream_ == Stream::Debug) {
+            // If we're writing to OpmLog::debug then continue to do so.
+            return;
+        }
+
+        this->log_function_ = new_stream;
+        (*this)(messages);
+    }
+
+    std::string format_message(const std::string& message) const
+    {
+        return fmt::format("{}{}", this->prefix_, message);
+    }
 };
 
+    ScheduleLogger::Stream select_stream(const bool log_to_debug,
```

Comment:
Static method or free function? Does not use anything of the instance of this class

Reply:
Or a lambda at the only place where it used?